### ISSUE TITLE
Sprint October 16th - October 30th

### ISSUE THREAD

**Issue by benjackwhite** [PostHog/posthog#17908]
# Global Sprint Planning

## 3 things that might take us down

1. Clickhouse visibility - had an incident, and then we had to sudo command in. 
2. 
3.

## Retro: What can we do better next sprint?

1. 
2.
3.
5.
6.

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by benjackwhite** [comment]
# Team Monitoring

**Support hero:** @benjackwhite 

## Retro



- @daibhin Not good to be struggling for a "thing of the week" - why was that?
- @pauldambra ingestion validation was painful as it led to bugs and hence incidents. Do we fix it? Do we hassle pipeline to fix it (strong opinions and knowledge in that team so well suited)?
- @benjackwhite Clickhouse visibility is really hard which meant we took waaaay longer to know
- @benjackwhite Polishing a lot on Notebooks - could we already just release?

## Hang over items from previous sprint



- Replays in Clickhouse deprecation 
  - Just waiting on big merge
- Console logs searchable - bit of a side quest

## OKR

📱 Mobile Replay 🟢
- Android SDK rewrite done ✅ 
- iOS rewrite underway ✅  

### High priority

- Release Notebooks! @benjackwhite 
  - why? We are approaching too much polish. There is already strong use cases so lets get it in front of more people
- Offsite hackathon - release or kill!
  - why? we have cool ideas already, try to make sure something ends up actually getting released
- iOS rewrite @marandaneto 

### Low priority / side quests
- Replay ingestion tidying

**Comment by fuziontech** [comment]
# Team Data Warehouse

### OKR Q4

### Objective

**PostHog users can effortlessly bring their business data to PostHog and analyze it** 🟢

* **Key Results:**
    * Be able to import data into data warehouse in one click from most popular SaaS tools (X, Y, Z connectors built)
    * 5 happy users/teams who have imported external data
    * Data Warehouse queries are compatible with insights hogQL

**PostHog is moved over to whatever CH infrastructure we will be using for the next 1 year (at least)** 🟢

* **Key Results:**
    * We are either on new infrastructure for EU that we host or on ClickHouse Cloud
    * Queries for large customers are performing faster
    * We can scale more easily (without James bespoke effort)
 

### Retro

**Product**
- [ ] convert revenue dashboard to posthog
- [ ] more user outreach

**Infra**
- [x] Finish all Vector setups (some are problematic)
- [x] Sync all tables to be = between environments
- [x] Optimize all tables
- [ ] Develop ability to feature flag / switch cluster a query is dispatched to based on team
- [x] Cut costs on s3 backups (TTL/Life cycling)


### High priority

**Product**
- [ ] Build and test stripe connector for one click data import/sync
- [ ] Work through pricing plan for data import. (Necessary because the costs could runaway if we don't charge from the onset) 

**Infra**
- [ ] Develop ability to feature flag / switch cluster a query is dispatched to based on team
- [ ] Further testing 720GiB x 3 cluster after downscaling
- [ ] Move over one team for testing 👀
- [ ] Cleanups from CH issues last week

**Comment by neilkakkar** [comment]
# Team Feature Success

**Support hero:** @jurajmajerik 

Days off:
Juraj: 1 day
Neil: 1 day
Li: 1 day
Annika: 1 day + offsite w Cory

## Retro

1. Survey results visualisation polish (based on Cory's mocks): @jurajmajerik -> 3/5 done, 1 in progress, 1 will roll over. Same for events overview.
2. Survey creation flow polish ([based on mocks](https://github.com/PostHog/posthog/issues/16018#issuecomment-1624362200)): @smallbrownbike (+ @neilkakkar on standby for any hogql / app support) - almost done
3. Tackle everything in https://github.com/PostHog/posthog/issues/17038 - @liyiy  ( + @neilkakkar ) - a bit too many clashes in here, worked on adjacent stuff (like templates, html support, and e2e tests), but full QA next sprint.

## Hang over items from previous sprint


---

## OKRs

1. Get 5 reference customers for surveys
2. Get reference customers for feature flags and experiments

### High priority

1. Surveys QA & polish before launch (https://github.com/PostHog/posthog/issues/17038): @liyiy / @neilkakkar 
2. Finish survey result visualisation updates & events overview: @jurajmajerik 

### Low priority / side quests

**Comment by robbie-c** [comment]
# Team Web Analytics

**Support hero:** n/a

## Hang over items from previous sprint
* Add filters to the web dashboard

## OKR
🟢 Build the dashboards to match 80% of functionality - work underway and making good progress
⚪  Get 5 20k+ customers to turn PostHog on on their website - not started
⚪ Start working towards the architecture to scale to a general release - not started

### High priority
* Continue adding filters
* Write more tests for the queries and filtering

### Low priority
* UI polish, to prevent the beta from making people's eyes bleed

**Comment by raquelmsmith** [comment]
# Growth

## Retro

- [x] Launch new onboarding experiment!
- [ ] Get surveys ready for pricing launch https://github.com/PostHog/billing/pull/366
  - in progress
- [ ] Fix inability to delete orgs due to foreign key on old billing tables
- [x] Figure out why usage reports have been failing
- [ ] Recover lost revenue for delayed reports
  - https://github.com/PostHog/product-internal/pull/514
  - PR for fix out this week
- [ ] Change MRR per product reporting to be calculated based on usage, not invoices

## OKR

- ⚪  Objective 1: Better understand what leads to revenue
  - Reasoning: We're still a bit fuzzy about what metrics are leading indicators for revenue.
  - KR: A company scorecard with actionable insights that help us understand where to invest
  - KR: An understanding (doc or dashboard) of how people use multiple products
- ⚪ Objective 2: Simplify and improve the upgrade process
  - Reasoning: It's currently somewhat confusing and requires too many clicks.
  - KR: Increase the percentage of orgs who upgrade to a paid account
  - KR: Decrease or maintain the number of surprise bills that CS has to refund
  - KR: Automatically subscribe people to the products they use
- ⚪ Bonus Objective 3: Continue experimenting with onboarding and activation metrics
  - KR: Product-specific onboarding performs better than the old onboarding flow
  - KR: 2 experiments for Product Analytics & Session Replay activation metrics each

## This sprint
### High priority
- [ ] Get surveys ready for pricing launch https://github.com/PostHog/billing/pull/366
- [ ] Fix inability to delete orgs due to foreign key on old billing tables
- [ ] Fix revenue forecasting for quarterly billing
- [ ] Keep iterating on onboarding exp until it's good

### Low priority / side quests

- [ ] Change MRR per product reporting to be calculated based on usage, not invoices

**Comment by mariusandra** [comment]
# Project Query Engine

## OKR

- Project "Finish the darned Query Engine" (Tom, Marius)
  - Finish insight HogQL conversion.
  - Support non-event data sources on HogQL insights (BI tools for data warehouse).
  - Build tools to measure HogQL query performance, and catch regressions before they hit users.

## Retro

- Finish the trends & lifecycle insights --> rollover
- Start the HogQL benchmarking suite --> did not start.
- Connect other data sources to the trends query --> did not start

## High priority

- Finish the trends insights, and related features (persons query from it, breakdown value query) @Gilbert09 
- Split PoE into "person properties on events" and "person id on events without an expensive join", and do the first part. @mariusandra 
- Start the Query Planner, integrate [new person/version join](https://github.com/PostHog/posthog/pull/17811) and [cohort optimisation](https://github.com/PostHog/posthog/pull/17354). @mariusandra 

## Low priority / side quests

- Take on the next insights (funnels, paths, retention)
- Connect other data sources to the trends query
- Async slow lane for queries that take 60 < x < 600 seconds to complete.

**Comment by timgl** [comment]
# Team Pipeline

**Support hero:** ___

## Retro


- batch exports: migrate BigQuery and PG users @tomasfarias
  - High frequency batch exports for a customer came up, they're happy with the test so they'll fully migrate next week
  - Made progress on various PRs (backfills, logs), just need to get them merged in
- Investigate Postgres issues related to PoE @bretthoerner 
  - Mostly worked on switching to RDKafka
  - We'll recreate the table with fewer constraints, turn it on for team 2 and see if it works

Other topics
- RDKafka for events ingestion completed @xvello 
- Fixed restarts
- Progress on capture v2
- Improved migration progress to cloud

## Hang over items from previous sprint



- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red

- keep the lights on
  - green, need to do the redis split before things go bad
- Batch exports migrations complete - @tomasfarias
  - green, on track to migrate everyone over
- Webhooks v2 @tiina303
  - New delivery system
    - red, we have not started work and we have no plan for it as we'll probably need to build a new system
  - New interface for webhooks and onevent apps
    - green, we've shipped plugin server types so webhooks are a universal thing

### High priority

- Capture-rs is deployed and capturing traffic for team 2 from PostHog-js @xvello 
- Make replicator use compose webhook and elements_chain @tiina303 
- Finish migrations of Bigquery and Postgres, and finish redshift app @tomasfarias 
- Run PoE experiment in prod and figure out next steps depending on that @bretthoerner 

### Low priority / side quests

- PostHog-js to send elements_chain as a strong @tiina303 
- Showing users webhook metrics @tiina303
- Work with Monitoring team to write mini RFC with shape of payload and failures modes of capture @xvello

**Comment by benjackwhite** [comment]
# Project Noteforce3000

## Hang over items from previous sprint

- Flag the new sidebar nav and default to more standard nav (posthog logo?) @Twixes 
  - 🟡 few things still to be done
- Finalise V1 of the topbar area (search button? contextual icons?) @benjackwhite    
  - 🔴 Didn't find a good approach. Will solve with @corywatilo on offsite
- Unified search (cmd+k does the resource search as well as quick actions) @thmsobrmlr 
  - 🟡 Half way done - lots of inspiration from posthog.com

## OKR

1. Project "Noteforce Exploration 3000" 
* Pair up with Team Monitoring to complete projects Data Exploration, Notebooks and PostHog 3000.
* Release new navigation and insight editor.
* Work more HogQL prompts into the interface.
* Build metrics into the system, introduce Metric Mondays.


### High priority

- Offsite
  - Concrete Roadmap (and a timeline) for "3000" (and a clear idea of what that is)
    - Very broken down, with a sense of realistic releasable chunks
    - Broken down into specific features
  - Clear who is building what, and how to collaborate on it to ensure things get shipped
  - Hackathon based on these outcomes

**Comment by ellie** [comment]
# Team Infrastructure

**Support hero:** @ellie

Daniel on PTO next week

## Hang over items from previous sprint



- Item 1. prioritised/deprioritise

## OKR

1. Continued maintenance of systems
2. Controlling costs
3. Platform improvements
4. Clickhouse changes

### High priority

- roll out new ALB to prod-us @frankh
- look into switching cloudfront -> cloudflare @frankh 
- enable rack-awareness for Clickhouse kafka @frankh
- rollout terragrunt in prod @danielxnj 
- deployment improvements and fixes @danielxnj 
- rollout capture-rs for team 2 @ellie 


### Low priority / side quests

-
